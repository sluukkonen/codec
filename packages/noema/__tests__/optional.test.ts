import { object, string, optional } from "../src/index.js"
import { expectParseFailure, expectParseSuccess } from "./helpers.js"

describe("optional", () => {
  test("should parse undefined or a value parsed by the underlying codec", () => {
    expectParseSuccess(optional(string), undefined)
    expectParseSuccess(optional(string), "")
  })

  test("should fail to parse other values", () => {
    expectParseFailure(optional(string), 1, [
      { code: "invalid_string", path: [] },
    ])
  })

  test("should work with object types", () => {
    expectParseSuccess(
      object({
        a: optional(string),
      }),
      {}
    )
  })
})
