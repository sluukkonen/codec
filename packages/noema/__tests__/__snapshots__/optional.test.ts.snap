// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`optional should fail to parse other values 1`] = `
[
  {
    "code": "invalid_string",
    "path": [],
  },
]
`;
